cloudinit/config/cc_ca_certs.py

```diff
@@ -107,88 +135,108 @@ def add_ca_certs(distro_cfg, certs):
     """
     if not certs:
         return
-    # First ensure they are strings...
-    cert_file_contents = "\n".join([str(c) for c in certs])
-    util.write_file(
-        distro_cfg["ca_cert_full_path"], cert_file_contents, mode=0o644
-    )
-    update_cert_config(distro_cfg)
+    # Write each certificate to a separate file, maximum 99 certificates.
+    cert_number = 0
+    for c in certs:
+        # First ensure they are strings...
+        cert_file_contents = str(c)
+        cert_number += 1
+        if cert_number == 100:
+            LOG.error(
+                "100+ certificates specified, ignoring more than the 1st"
+                " 99 certs."
+            )
+            return
+
+        cert_file_name = distro_cfg["ca_cert_full_path"] % {
+            "number": str(cert_number).zfill(2)
+        }
+        util.write_file(cert_file_name, cert_file_contents, mode=0o644)
+
+
+def disable_default_ca_certs(distro_name, distro_cfg):
+    """
+    Disables all default trusted CA certificates. To actually apply the
+    changes you must also call L{update_ca_certs}.
+
+    @param distro_name: String providing the distro class name.
+    @param distro_cfg: A hash providing _distro_ca_certs_configs function.
+    """
+    if distro_name == "freebsd":
```

Comment:
and here

Reply:
I didn't look at any of the other BSDs, do they use the same mechanism/utilities for managing certificates?

Reply:
Good point. dragonfly is one major release behind. but that still means it should be available.

Reply:
ah, "freebsd" is a family, I'd assumed there was a single "bsd" family defined. Yes this could use family rather than distro_name, though when I checked in the tests thereis currrently nothing in place to handle osfamily lookups so I'd have to figure that out.

Reply:
Removed FreeBSD related changes from this PR.